### PR TITLE
Deprecate tape.unwrap

### PR DESCRIPTION
**Pull request by lillian542**
Deprecate `tape.unwrap` and promote `convert_to_numpy_parameters` as an alternative

### CODE REVIEW

**Comment by lillian542** [comment]
[sc-38655]

File: doc/development/deprecations.rst [hunk 1]
```diff
@@ -54,6 +54,25 @@ Pending deprecations
...
+
+  Instead of ``tape.unwrap()``, use :func:`~.transforms.convert_to_numpy_parameters`:
+
+  .. code-block:: python
+
+    from pennylane.transforms import convert_to_numpy_parameters
+
+    qscript = qml.tape.QuantumScript([qml.RX(torch.tensor(0.1234), 0)],
+                                     [qml.expval(qml.Hermitian(torch.eye(2), 0))] )
+    unwrapped_qscript = convert_to_numpy_parameters(circuit)
```
Comment (timmysilv) [inline doc/development/deprecations.rst:71]:
```suggestion
    unwrapped_qscript = convert_to_numpy_parameters(qscript)
```

File: pennylane/tape/qscript.py [hunk 2]
```diff
@@ -1156,6 +1157,12 @@ def unwrap(self):
         <tf.Tensor: shape=(), dtype=float32, numpy=0.2>,
         <tf.Variable 'Variable:0' shape=() dtype=float32, numpy=0.3>]
         """
+
+        warnings.warn(
+            "The method tape.unwrap is deprecated and will be removed in PennyLane v0.33. "
+            "Please use convert_to_numpy_parameters instead."
```
Comment (timmysilv) [inline pennylane/tape/qscript.py:1163]:
```suggestion
            "Please use qml.transforms.convert_to_numpy_parameters instead."
```

File: pennylane/tape/unwrap.py [hunk 3]
```diff
@@ -119,6 +120,11 @@ class UnwrapTape:
     """
 
     def __init__(self, tape, params=None):
+        warnings.warn(
+            "The UnwrapTape class is deprecated and will be removed in PennyLane v0.33. "
+            "Please use convert_to_numpy_parameters instead."
```
Comment (timmysilv) [inline pennylane/tape/unwrap.py:125]:
```suggestion
            "Please use qml.transforms.convert_to_numpy_parameters instead."

```

File: doc/releases/changelog-dev.md
Comment (timmysilv) [inline doc/releases/changelog-dev.md:1]: don't forget the PR link, and adding yourself as a contributor!

**Review by timmysilv — APPROVED** [review]
a few lil things, but otherwise it's good to go from my POV!

**Review by mudit2812 — APPROVED** [review]
Looks good! Just wanted to reiterate Matt's point of adding yourself as a contributor and the PR link.

**Review by mudit2812 — CHANGES_REQUESTED** [review]
Just going to mark as changes requested so that the small changes aren't forgotten :)

File: doc/development/deprecations.rst [hunk 4]
```diff
@@ -54,6 +54,25 @@ Pending deprecations
   - Deprecated in v0.32
   - Will be removed in v0.33
 
+* The method ``tape.unwrap()`` and corresponding ``UnwrapTape`` class are deprecated, and
```
Comment (albi3ro) [inline doc/development/deprecations.rst:57]: Do we also need to explicitly deprecate `qml.tape.Unwrap`? Forgot they were different classes. @trbromley

Comment (albi3ro) [reply doc/development/deprecations.rst:69]: nope.  it is a slight variation on `tape.unwrap` and `UnwrapTape`.

Comment (lillian542) [reply doc/development/deprecations.rst]: Added deprecation for `Unwrap` :)

**Review by mudit2812 — APPROVED** [review]
🚀

**Review by albi3ro — APPROVED** [review]
👍

**Review by trbromley — APPROVED** [review]

**Pull request merged**